lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -112,20 +112,29 @@ public function get(QueryCacheKey $key, ResultSetMapping $rsm, array $hints = ar
         $regionName  = $region->getName();
 
         $cm = $this->em->getClassMetadata($entityName);
+
+        $entityCacheKeys = [];
+        foreach ($entry->result as $index => $concreteEntry) {
+            $entityCacheKeys[$index] = new EntityCacheKey($cm->rootEntityName, $concreteEntry['identifier']);
+        }
+        $entries = $region->getMultiple(new CollectionCacheEntry($entityCacheKeys));
```

Comment:
@Ocramius does it makes sense to extract this process to a named constructor on `CollectionCacheEntry`?